server/models/SiteMetadataModel/index.js

```diff
@@ -1,15 +1,21 @@
 import { collections } from '../../utils/mongoCollections'
 
+const queryOptions = {
+  upsert: true,
+  returnDocument: 'after',
+}
 const SiteMetadataModel = {
   findOne: async (db, query) =>
     await db.collection(collections.metadata).findOne(query),
-  upsert: async (db, query, metadataAttributes) =>
-    await db
-      .collection(collections.metadata)
-      .findOneAndUpdate(query, metadataAttributes, {
-        upsert: true,
-        returnDocument: 'after',
-      }),
+  upsert: async (db, query, { setAttributes, addToSetAttributes }) =>
+    await db.collection(collections.metadata).findOneAndUpdate(
+      query,
+      {
+        $set: { ...(setAttributes || {}), updatedAt: new Date() },
+        $addToSet: addToSetAttributes || {},
```

Comment:
```suggestion
        $addToSet: { ...(setAttributes || {}), createdAt: new Date() },
```

Reply:
we need both operators

Reply:
right, but should we set the createdAt timestamp here?

Reply:
yes, I think it's a good idea to add a createdAt timestamp to the new elements that are appended to the createdAt. There were some mongo errors when I wanted to use this shorthand, and I had to be much more verbose to achieve this. 

[6263301](https://github.com/AMP-SCZ/dpdash/pull/723/commits/6263301c5604f13919f33d96f3a423b5a2199d7d)